src/services/invokeService.ts

```diff
@@ -0,0 +1,86 @@
+import { BaseService } from "./baseService"
+import Serverless from "serverless";
+import config from "../config";
+import axios from "axios";
+import { FunctionAppService } from "./functionAppService";
+import { FunctionAppResource } from "../armTemplates/resources/functionApp";
+import { ServerlessAzureConfig } from "../models/serverless";
+
+export class InvokeService extends BaseService {
+  public serverless: Serverless;
+  public options: Serverless.Options;
+  public constructor(serverless: Serverless, options: Serverless.Options) {
+    super(serverless, options);
+    this.serverless = serverless;
+    this.options = options;
+  }
+
+  /**
+   * Invoke an Azure Function
+   * @param functionName Name of function to invoke
+   * @param data Data to use as body or query params
+   * @param method GET or POST
+   */
+  public async invoke(functionName: string, data: any, method: string){
+    /* accesses the admin key */
+    if (!(functionName in this.slsFunctions())) {
+      this.serverless.cli.log(`Function ${functionName} does not exist`);
+      return;
+    }
+    const functionObject = this.slsFunctions()[functionName];
+    const eventType = Object.keys(functionObject["events"][0])[0];
+    if (eventType !== "http") {
+      this.log("Needs to be an http function");
+      return;
+    }
+
+    let url = `http://${FunctionAppResource.getResourceName(this.serverless.service as any)}${config.functionAppDomain}${config.functionAppApiPath + functionName}`;
+
+    if (method === "GET") {
+      const queryString = this.getQueryString(data);
+      url += `?${queryString}`
+    }
+
+    this.log(url);
+    const options = await this.getOptions(method, data);
+    this.log(`Invoking function ${functionName} with ${method} request`);
+    return await axios(url, options);
+  }
+
+  private getQueryString(eventData: any) {
+    if (typeof eventData === "string") {
+      try {
+        eventData = JSON.parse(eventData);
+      }
+      catch (error) {
+        return Promise.reject("The specified input data isn't a valid JSON string. " +
+          "Please correct it and try invoking the function again.");
+      }
+    }
+    return Object.keys(eventData)
+      .map((key) => `${key}=${eventData[key]}`)
+      .join("&");
+  }
+
+  /**
+   * Get options object  
+   * @param method The method used (POST or GET)
+   * @param data Data to use as body or query params
+   */
+  private async getOptions(method: string, data?: any) {
+    
+    const functionAppService = new FunctionAppService(this.serverless, this.options);
+    const functionsAdminKey = await functionAppService.getMasterKey();
+    const options: any = {
+      host: config.functionAppDomain,
```

Comment:
Same as above - You'll need to retrieve the function app domain from the function app dynamically.